### PR TITLE
Cleanup e2e tests after introduction of kubernetes v1.29

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
This PR is a follow up to address most of the comments from #3048

**What type of PR is this?**
/kind cleanup

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by kron4eg** [comment]
lol, seems like I've removed 1.29 too :D

File: test/tests.yml [hunk 1]
```diff
@@ -183,21 +135,21 @@
...
-    - name: aws_centos_stable
-    - name: aws_default_stable
-    - name: aws_flatcar_stable
-    - name: aws_rhel_stable
-    - name: aws_rockylinux_stable
+    - name: azure_centos_stable
+    - name: azure_default_stable
+    - name: azure_flatcar_stable
+    - name: azure_rhel_stable
+    - name: azure_rockylinux_stable
```
Comment (xmudrii) [inline test/tests.yml:152]: This scenario is not possible because there are no in-tree CCMs in v1.29.

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: test/tests.yml (hunk 1 above)
Comment (kron4eg) [reply test/tests.yml]: Dropped!

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

File: test/tests.yml [hunk 2]
```diff
@@ -183,21 +135,11 @@
...
```
Comment (xmudrii) [inline test/tests.yml:142]: Just realized we're missing GCE here. Can you quickly add it?

**Review by xmudrii — CHANGES_REQUESTED** [review]
/hold
@kron4eg PTAL at this comment

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**